### PR TITLE
use nodeMap for predicates

### PR DESCRIPTION
**Pull request by njlkj**
#1231 
/assign @william-wang

### CODE REVIEW

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 1]
```diff
@@ -131,8 +129,8 @@ func calculateWeight(args framework.Arguments) priorityWeight {
 func (pp *nodeOrderPlugin) OnSessionOpen(ssn *framework.Session) {
 	weight := calculateWeight(pp.pluginArguments)
 	pl := util.NewPodLister(ssn)
-	pods, _ := pl.List(labels.NewSelector())
-	nodeMap, nodeSlice := util.GenerateNodeMapAndSlice(ssn.Nodes)
+	//pods, _ := pl.List(labels.NewSelector())
+	nodeMap := util.GenerateNodeMapAndSlice(ssn.Nodes)
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/nodeorder/nodeorder.go:133]: Why remove pods?

**Comment by Thor-wl** [comment]
The motivation of this modification is?

File: pkg/scheduler/plugins/nodeorder/nodeorder.go (hunk 1 above)
Comment (njlkj) [reply pkg/scheduler/plugins/nodeorder/nodeorder.go:133]: ssn.Nodes has all pods info , and will be updated after one task is finished. 
But if use relist pods, with gang scheduler, podAntiAffinity not take effect.In the same job, two anti-affinity PODs scheduled on the same node.
```
---
apiVersion: scheduling.volcano.sh/v1beta1
kind: PodGroup
metadata:
  name: test
  namespace: default
spec:
  minMember: 2
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-nginx
spec:
  selector:
    matchLabels:
      app: my-nginx
  replicas: 2
  template:
    metadata:
      labels:
        app: my-nginx
      annotations:
        scheduling.k8s.io/group-name: test
    spec:
      schedulerName: volcano
      containers:
      - name: my-nginx
        image: nginx
        ports:
        - containerPort: 80
        resources:
          requests:
            memory: "1Gi"
            cpu: "1000m"
          limits:
            memory: "1Gi"
            cpu: "1000m"
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            - labelSelector:
                matchExpressions:
                  - key: app
                    operator: In
                    values:
                      - "my-nginx"
              topologyKey: "kubernetes.io/hostname"
```

File: Makefile [hunk 2]
```diff
@@ -83,7 +83,7 @@ image_bins: init
 	CC=${CC} CGO_ENABLED=1 $(GOBIN)/gox -osarch=${REL_OSARCH} -ldflags ${LD_FLAGS} -output ${BIN_DIR}/${REL_OSARCH}/vc-scheduler ./cmd/scheduler;
 
 images: image_bins
-	for name in controller-manager scheduler webhook-manager; do\
```
Comment (Thor-wl) [inline Makefile:86]: Why do the modification?

File: installer/dockerfile/scheduler/Dockerfile [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 
-FROM alpine:latest
+FROM centos:latest
```
Comment (Thor-wl) [inline installer/dockerfile/scheduler/Dockerfile:16]: why change the base image?

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 4]
```diff
@@ -88,7 +88,7 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
 		}
 	}
 
-	klog.V(3).Infof("Try to enqueue PodGroup to %d Queues", len(jobsMap))
+	klog.V(3).Infof("Try to enqueue PodGroup to %d Queues", queues.Len())
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:91]: This queues is `PriorityQueue` , not CRD `Queue`. Please reset to original

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 5]
```diff
@@ -105,8 +105,11 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
 	}
 	idle := total.Clone().Multi(enqueue.getOverCommitFactor(ssn)).Sub(used).Sub(lockedNodesIdle)
 
+	klog.V(3).Infof("Try to enqueue PodGroup to %d Queues", queues.Len())
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:108]: Repetitive log

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 6]
```diff
@@ -105,8 +105,11 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
...
 	for {
+		klog.V(3).Infof("Try to enqueue PodGroup to %d Queues", queues.Len())
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:110]: The same as above

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 7]
```diff
@@ -105,8 +105,11 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
...
 		if queues.Empty() {
+			klog.V(3).Infof("queues is Empty, ignore it.")
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:112]: Meaningless log

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 8]
```diff
@@ -130,9 +133,12 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
 
 		inqueue := false
 
+		klog.V(4).Info("debug JobEnqueueable")
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:136]: Please remove debug log

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 9]
```diff
@@ -130,9 +133,12 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
...
 		if job.PodGroup.Spec.MinResources == nil {
+			klog.V(4).Info("debug not JobEnqueueable")
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:138]: ditto

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 10]
```diff
@@ -130,9 +133,12 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
...
 			inqueue = true
 		} else {
+			klog.V(4).Info("debug JobEnqueueable")
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:141]: ditto

File: pkg/scheduler/framework/session_plugins.go [hunk 11]
```diff
@@ -284,10 +285,13 @@ func (ssn *Session) JobValid(obj interface{}) *api.ValidateResult {
 func (ssn *Session) JobEnqueueable(obj interface{}) bool {
 	for _, tier := range ssn.Tiers {
 		for _, plugin := range tier.Plugins {
+			klog.V(4).Info("debug JobEnqueueable plugin")
```
Comment (Thor-wl) [inline pkg/scheduler/framework/session_plugins.go:288]: ditto

File: pkg/scheduler/framework/session_plugins.go [hunk 12]
```diff
@@ -284,10 +285,13 @@ func (ssn *Session) JobValid(obj interface{}) *api.ValidateResult {
...
 			fn, found := ssn.jobEnqueueableFns[plugin.Name]
 			if !found {
+				klog.V(4).Info("debug JobEnqueueable plugin not found")
```
Comment (Thor-wl) [inline pkg/scheduler/framework/session_plugins.go:291]: ditto

File: pkg/scheduler/framework/session_plugins.go [hunk 13]
```diff
@@ -284,10 +285,13 @@ func (ssn *Session) JobValid(obj interface{}) *api.ValidateResult {
...
 				continue
 			}
+			klog.V(4).Info("debug JobEnqueueable plugin  found")
```
Comment (Thor-wl) [inline pkg/scheduler/framework/session_plugins.go:294]: ditto

File: pkg/scheduler/plugins/proportion/proportion.go [hunk 14]
```diff
@@ -260,6 +259,7 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 	})
 
 	ssn.AddJobEnqueueableFn(pp.Name(), func(obj interface{}) bool {
+		klog.V(4).Info("debug add AddJobEnqueueableFn")
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/proportion/proportion.go:262]: remove debug log

File: pkg/scheduler/plugins/proportion/proportion.go [hunk 15]
```diff
@@ -279,6 +279,8 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 		minReq := GetJobMinResources(job.PodGroup.Spec)
 		// The queue resource quota limit has not reached
 		inqueue := minReq.Add(attr.allocated).Add(attr.inqueue).LessEqual(api.NewResource(queue.Queue.Spec.Capability))
+		klog.V(4).Infof("debug allocated %v inqueue %v minReq %v Capability %v", attr.allocated, attr.inqueue, minReq, api.NewResource(queue.Queue.Spec.Capability))
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/proportion/proportion.go:282]: +1

File: pkg/scheduler/plugins/proportion/proportion.go [hunk 16]
```diff
@@ -279,6 +279,8 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
...
+		klog.V(4).Infof("debug LessEqual %v", inqueue)
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/proportion/proportion.go:283]: +1

Comment (njlkj) [reply pkg/scheduler/plugins/proportion/proportion.go:283]: sorry , I made a mistake

File: pkg/scheduler/plugins/proportion/proportion.go (hunk 14 above)
Comment (njlkj) [reply pkg/scheduler/plugins/proportion/proportion.go:262]: sorry , I made a mistake

File: pkg/scheduler/framework/session_plugins.go (hunk 12 above)
Comment (njlkj) [reply pkg/scheduler/framework/session_plugins.go:291]: sorry , I made a mistake

Comment (njlkj) [reply pkg/scheduler/framework/session_plugins.go]: fixed

File: pkg/scheduler/framework/session_plugins.go (hunk 13 above)
Comment (njlkj) [reply pkg/scheduler/framework/session_plugins.go]: fixed

File: pkg/scheduler/plugins/proportion/proportion.go (hunk 14 above)
Comment (njlkj) [reply pkg/scheduler/plugins/proportion/proportion.go]: fixed

File: pkg/scheduler/plugins/proportion/proportion.go (hunk 15 above)
Comment (njlkj) [reply pkg/scheduler/plugins/proportion/proportion.go]: fixed

File: pkg/scheduler/plugins/proportion/proportion.go (hunk 16 above)
Comment (njlkj) [reply pkg/scheduler/plugins/proportion/proportion.go]: fixed

File: pkg/scheduler/actions/enqueue/enqueue.go (hunk 6 above)
Comment (njlkj) [reply pkg/scheduler/actions/enqueue/enqueue.go]: fixed

File: pkg/scheduler/actions/enqueue/enqueue.go (hunk 4 above)
Comment (njlkj) [reply pkg/scheduler/actions/enqueue/enqueue.go]: fixed

File: Makefile (hunk 2 above)
Comment (njlkj) [reply Makefile]: fixed

File: installer/dockerfile/scheduler/Dockerfile (hunk 3 above)
Comment (njlkj) [reply installer/dockerfile/scheduler/Dockerfile]: fixed

**Comment by Thor-wl** [comment]
I have no questions after your fix to first review. Please fix the DCO and i will tag `approve`.

File: pkg/scheduler/plugins/util/k8s/snapshot.go [hunk 17]
```diff
@@ -51,8 +51,7 @@ func NewEmptySnapshot() *Snapshot {
 }
 
 // NewSnapshot initializes a Snapshot struct and returns it.
-func NewSnapshot(pods []*v1.Pod, nodes []*v1.Node) *Snapshot {
-	nodeInfoMap := createNodeInfoMap(pods, nodes)
```
Comment (shinytang6) [inline pkg/scheduler/plugins/util/k8s/snapshot.go:55]: The root cause is that original `nodeInfoMap` was not updated in one scheduling cycle, right?

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 18]
```diff
@@ -131,8 +129,8 @@ func calculateWeight(args framework.Arguments) priorityWeight {
 func (pp *nodeOrderPlugin) OnSessionOpen(ssn *framework.Session) {
 	weight := calculateWeight(pp.pluginArguments)
 	pl := util.NewPodLister(ssn)
-	pods, _ := pl.List(labels.NewSelector())
-	nodeMap, nodeSlice := util.GenerateNodeMapAndSlice(ssn.Nodes)
+	//pods, _ := pl.List(labels.NewSelector())
```
Comment (shinytang6) [inline pkg/scheduler/plugins/nodeorder/nodeorder.go:132]: Then this comment can be removed.

Comment (Thor-wl) [reply pkg/scheduler/plugins/nodeorder/nodeorder.go:132]: Please fix the DCO

**Comment by Thor-wl** [comment]
Please resolve the conflicts

**Pull request closed without merging**